src/components/Grid/snapping.js

```diff
@@ -76,24 +76,36 @@ export function expandWindowAlongEdge(edge, center, { width, spacing, window_def
 }
 
 export function findClosestEdge(edges, cursor) {
-  const
-    withDistance = edges.map(e => ({
-      ...e,
-      ...pointDistanceToSegment(cursor, { start: e.v1, end: e.v2 }),
-    }));
+  // Use two references to defer creation of the object until the final return
+  let edge = undefined;
+  let dist = undefined;
+  let smallest = Infinity;
+  for (let e of edges) {
+    const d = pointDistanceToSegment(cursor, { start: e.v1, end: e.v2 });
+
+    if (d.dist < smallest) {
+      // Negative distances aren't possible, so getting a 0 is assured to be the closest
+      if (d.dist === 0) {
+        return {
+          ...e,
+          ...d,
+        };
+      }
 
-  let ret = undefined;
-  let smallest = NaN;
-  withDistance.forEach((d) => {
-    // The conditional is intentionally convoluted - simplifying it to just '' will have issues because it starts as NaN
-    // All comparisons to NaN return false, so for the first iteration it has to negate it
-    if (!(d.dist >= smallest)) {
       smallest = d.dist;
-      ret = d;
+      edge = e;
+      dist = d;
     }
-  });
+  }
 
-  return ret;
+  if (edge) {
+    return {
+      ...edge,
+      ...dist,
+    };
+  } else {
+    return undefined;
```

Comment:
I notice you're setting variables to undefined above, and returning it here as well. I was always taught to avoid explicitly using `undefined`. To quote a book I have "Comparing directly against undefined is unsafe because you rely on permitted but discouraged practices." 

I'm wondering if we could define some sort of default edge / dist and use that for comparison here. Additionally, other pieces of code relying on `findClosestEdge` could do more distinct checks on the return value. We could even throw an error here if returning undefined should technically be unreachable.

Reply:
For the most part, the goal is to mimic the behavior of [_.minBy](https://lodash.com/docs/4.17.15#minBy), which was the original implementation of the function.  When given an empty array, it returns `undefined`.

Since it's just performance changes, I'm disinclined to modify function signatures.

Reply:
Works for me.